### PR TITLE
feat: added GitLab container scanning report parser

### PR DESCRIPTION
**Pull request by natebwangsut**
Adding GitLab container scanning report parser

- https://docs.gitlab.com/ee/user/application_security/container_scanning

### CODE REVIEW

**Comment by natebwangsut** [comment]
@madchap @damiencarol

File: dojo/tools/gitlab_container_scan/parser.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            # Add mitigation if possible
+            if solution != "":
+                finding.mitigation = solution
+
+            # internal de-duplication via description + title
+            dupe_key = hashlib.sha256(str(description + title).encode('utf-8')).hexdigest()
+            if dupe_key in dupes:
+                find = dupes[dupe_key]
+                if finding.description:
+                    find.description += "\n" + finding.description
```
Comment (damiencarol) [inline dojo/tools/gitlab_container_scan/parser.py:93]: You could use the syntax `"\n\n-----\n\n" it produces horizontal lines.

**Review by damiencarol — CHANGES_REQUESTED** [review]

**Comment by natebwangsut** [comment]
@damiencarol I've commited the changes per your request, please let me know if you've got anything to add

File: dojo/tools/gitlab_container_scan/parser.py [hunk 2]
```diff
@@ -69,28 +60,35 @@ def get_findings(self, file, test):
...
             # Add component fields if not empty
+            if cve != "":
+                finding.cve = cve
+            if cwe != "":
+                finding.cwe = cwe
             if component_name != "":
-                finding.component_name = component_name[:190] + (component_name[190:] and '...')
+                finding.component_name = component_name[:190] + (
+                    component_name[190:] and "..."
+                )
```
Comment (damiencarol) [inline dojo/tools/gitlab_container_scan/parser.py:75]: For this kind of shortening task, you can use `textwrap.shorten` see https://docs.python.org/3/library/textwrap.html#textwrap.shorten

File: dojo/tools/gitlab_container_scan/parser.py [hunk 3]
```diff
@@ -26,40 +26,31 @@ def get_findings(self, file, test):
...
-                    if id['type'] == "cwe":
-                        cwe = id['value']
-                except:
-                    pass
+            for id in vulnerability["identifiers"]:
+                if "type" in id:
+                    if id.get("type") == "cve":
+                        cve = id["value"]
+                    if id.get("type") == "cwe":
+                        cwe = id["value"]
```
Comment (damiencarol) [inline dojo/tools/gitlab_container_scan/parser.py]: move this block to line 68, it will be clearer to read.

**Review by damiencarol — CHANGES_REQUESTED** [review]

**Comment by natebwangsut** [comment]
I've update the code per formatting

**Review by Maffooch — APPROVED** [review]

**Comment by natebwangsut** [comment]
@damiencarol should I apply `unique_id_from_tool` here as well ?

**Comment by damiencarol** [comment]
@natebwangsut yes for this parser.

**Comment by natebwangsut** [comment]
@Maffooch @damiencarol I've added unique ID for parsing + rebased ontop of latest dev as well, kindly review the PR

**Review by damiencarol — APPROVED** [review]

**Pull request merged**